include/flox/resolver/state.hh

```diff
@@ -50,68 +41,50 @@ class ResolverState : protected pkgdb::PkgDbRegistryMixin {
 
   protected:
 
+    /* From `PkgDbRegistryMixin':
+     *   std::shared_ptr<nix::Store>                         store
+     *   bool                                                force    = false
+     *   std::shared_ptr<Registry<pkgdb::PkgDbInputFactory>> registry
+     */
+
       [[nodiscard]]
-      virtual std::vector<std::string> &
+      std::vector<std::string> &
     getSystems() override
     {
       return this->preferences.systems;
     }
 
 
-    /**
-     * @brief Initializes `PkgDbRegistryMixin::registry` and scrapes inputs
-     *        when necessary.
-     */
-      void
-    initResolverState()
-    {
-      if ( this->registry == nullptr )
-        {
-          this->initRegistry();
-          this->scrapeIfNeeded();
-        }
-      assert( this->registry != nullptr );
-    }
-
-
   public:
 
-    ResolverState( const RegistryRaw             & registry
-                 , const pkgdb::QueryPreferences & preferences
+    ResolverState( RegistryRaw             registry
+                 , pkgdb::QueryPreferences preferences
                  )
-      : registryRaw( registry )
-      , preferences( preferences )
+      : registryRaw( std::move( registry ) )
```

Comment:
Was this previously doing a copy instead of a move?

Reply:
Honestly it was just a `clang-tidy` recommendation, so in this case I get to link you to a nice blurb explaining :)
https://clang.llvm.org/extra/clang-tidy/checks/modernize/pass-by-value.html